### PR TITLE
Refactor/dedup protobuf

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
Extract the protobuf definition files (curp/proto and xlineapi/proto) as two submodules, and reuse them between xline server and xline client sdk.

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Review by themanforfree — APPROVED** [review]

**Comment by bsbds** [comment]
Some of the proto files that will not be used in client sdk are also moved to external submodules. 
For example, client will not need `AppendEntries` in `message.proto`
```
    rpc AppendEntries(AppendEntriesRequest) returns (AppendEntriesResponse);
```

Should we keep these server only proto definitions?

**Review by iGxnon — COMMENTED** [review]
LGTM

**Comment by Phoenix500526** [comment]
> Some of the proto files that will not be used in client sdk are also moved to external submodules. For example, client will not need `AppendEntries` in `message.proto`
> 
> ```
>     rpc AppendEntries(AppendEntriesRequest) returns (AppendEntriesResponse);
> ```
> 
> Should we keep these server only proto definitions?

Done

**Review by themanforfree — APPROVED** [review]
LGTM

File: curp/src/lib.rs [hunk 1]
```diff
@@ -157,11 +157,11 @@
 )]
 
 pub use curp_external_api::LogIndex;
-pub use rpc::ProtocolServer;
 /// Expose for madsim simulation tests.
 pub use rpc::{
     protocol_client, FetchLeaderRequest, FetchLeaderResponse, ProposeRequest, ProposeResponse,
 };
+pub use rpc::{InnerProtocolServer, ProtocolServer};
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply curp/src/lib.rs]: Done

**Review by themanforfree — APPROVED** [review]

**Pull request merged**